### ISSUE TITLE
Dashboard does not work if user has data for some days but not two weeks

### ISSUE THREAD

**Issue by shankari** [e-mission/e-mission-phone#148]
The current check for user and aggregate information in the dashboard assumes either the user has no data, or has the full two weeks of data. But this can be incorrect when the user has been using the app for less than 2 weeks, i.e in the days right after sign up.

e.g. note that the user has 7 days worth of data while the aggregate has 14 days.

<img width="863" alt="screen shot 2016-10-02 at 1 42 09 pm" src="https://cloud.githubusercontent.com/assets/2423263/19023565/183a1b76-88a6-11e6-83ad-f556b94e963e.png">

**Comment by shankari** [comment]
Examples of results showing this behavior - can be unzipped and used to test the fix.
[results_dump.json.zip](https://github.com/e-mission/e-mission-phone/files/505242/results_dump.json.zip)

**Comment by shankari** [comment]
@sunil07t this needs to be fixed right away.

**Comment by shankari** [comment]
Here's an example of how the slice fails in this case. twoWeeksAgoDuration has length 8, userDuration has length 0. So all metrics are zero.

```
> twoWeeksAgoDuration
< [Object, Object, Object, Object, Object, Object] (6) = $8
> userDuration
< [] (0) = $9
```

**Comment by shankari** [comment]
| No data  | No Data | 
| ----- | ----- |
| ![simulator screen shot oct 2 2016 3 17 01 pm](https://cloud.githubusercontent.com/assets/2423263/19024207/609da1b4-88b3-11e6-9e1e-7510b8ef4bb6.png) | ![simulator screen shot oct 2 2016 3 17 09 pm](https://cloud.githubusercontent.com/assets/2423263/19024208/60a1b7b8-88b3-11e6-8deb-621cd3b6a916.png) |

**Comment by shankari** [comment]
Fixed by #149 and #151
